misc/language-tour-guide/src/components/code-editor/CodeEditor.tsx

```diff
@@ -4,18 +4,25 @@ import { solarizedLight } from "@uiw/codemirror-theme-solarized";
 
 import "./CodeEditor.styles.css";
 import { useCodeEditorStore } from "../../store/codeEditor";
+import { useLocation } from "react-router";
 
 export function CodeEditor() {
+  const { pathname } = useLocation();
   const code = useCodeEditorStore((state) => state.code);
   const setCode = useCodeEditorStore((state) => state.setCode);
 
+  const handleChangeCode = (value: string) => {
+    localStorage.setItem(`code-${pathname}`, value);
```

Comment:
I feel like we should hash this to have shorter keys. Just a thought.

We can also consider to store it in Zustand.